src/Microsoft.OpenApi/Services/OpenApiVisitorBase.cs

```diff
@@ -15,7 +15,21 @@ namespace Microsoft.OpenApi.Services
     public abstract class OpenApiVisitorBase
     {
         private readonly Stack<string> _path = new Stack<string>();
-       
+        private CurrentKeys _currentKeys = new CurrentKeys();
+
+
+
+        /// <summary>
+        /// Properties available to identify context of where an object is within OpenAPI Document
+        /// </summary>
+        public CurrentKeys CurrentKeys
+        {
+            get
+            {
+                return _currentKeys;
+            }
+        }
```

Comment:
Does this syntactic sugar work?

public CurrentKeys CurrentKeys { get; } = new CurrentKeys();

No need to list out the private field directly.

Reply:
Yeah, that will work.

My particular usecase is fairly obscure.  When we import an OpenAPI document into API Management, we take any inline schemas and move them into the components/schemas section so that we can store all of the schemas as a single JSON Schema document.  To do that we need to generate a name for those schemas.  In order to generate a unique name it is important to know the path/operation/response code/mediatype that the schema lives under.

More generically, I could imagine tooling that walks the an OpenAPIDocument and checks schemas for certain characteristics.  Knowing if it is a schema under a Request or a Response is very useful.  Knowing if it is a schema for a 500 or a 200 might be useful.  Knowing the media type of an example if you want to verify examples would be useful.

Reply:
Interesting... Technically, you can use the path stack to get that unique key for your use case I think. But having the keys in strong-typed object like this may prove useful in the future. I'll sign off.

---
In reply to: [194930721](https://github.com/Microsoft/OpenAPI.NET/pull/280#discussion_r194930721) [](ancestors = 194930721)

Reply:
Yeah, that's what I have done up to now.  However, it doesn't allow me to mimic the previous behaviour of our import for V2 OpenAPI and I do thing there is value in being able to get at these discrete pieces of context information.,

---
In reply to: [194931324](https://github.com/Microsoft/OpenAPI.NET/pull/280#discussion_r194931324) [](ancestors = 194931324,194930721)